documentation/aws-roles-and-cli-tools.md

```diff
@@ -0,0 +1,199 @@
+
+Install AWS CLI v2
+```bash
+curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
+unzip -q awscliv2.zip -d /tmp
+sudo /tmp/aws/install
+```
+Install AWS Vault
+```bash
+wget https://github.com/99designs/aws-vault/releases/download/v6.2.0/aws-vault-linux-amd64
+sudo mv ./aws-vault-linux-amd64 /usr/local/bin/aws-vault
+sudo chmod +x /usr/local/bin/aws-vault
+```
+Install GPG and Pass
+```bash
+sudo apt-get install -y gnupg pass
+```
+
+There's an excellent guide to [managing password with GPG, the command line and Pass](https://www.thepolyglotdeveloper.com/2018/12/manage-passwords-gpg-command-line-pass/) but the essentials are:
+
+#### Generate GPG Key
+
+```bash
+gpg --full-generate-key
+```
+Answer the interactive quetsions. I chose
+```
+Please select what kind of key you want:
+   (1) RSA and RSA (default)
+RSA keys may be between 1024 and 4096 bits long.
+What keysize do you want? (3072)
+    3072
+Please specify how long the key should be valid.
+         0 = key does not expire
+      <n>  = key expires in n days
+      <n>w = key expires in n weeks
+      <n>m = key expires in n months
+      <n>y = key expires in n years
+Key is valid for? (0)
+    2y
+````
+You are then prompted for some personal details
+```
+GnuPG needs to construct a user ID to identify your key.
+
+Real name: <YOUR-NAME>
+Email address: <YOUR-NAME>@digital.education.gov.uk
+Comment: Work
+You selected this USER-ID:
+    "<YOUR-NAME> (Work) <<YOUR-NAME>@digital.education.gov.uk>"
+```
+You are then prompted for a passphrase to protect the GPG key. I used my Windows password manager to generate and store the passphrase
+
+After moving the mouse to generate random bytes for prime generation, we see:
+```
+gpg: key 8C6B1A2FA5910DE0 marked as ultimately trusted
+gpg: revocation certificate stored as '/home/nrubuntu/.gnupg/openpgp-revocs.d/A5896473017A0DD9B983A03D8C6B1A2FA5910DE0.rev'
+public and secret key created and signed.
+
+pub   rsa3072 2021-02-11 [SC] [expires: 2023-02-11]
+      A5896473017A0DD9B983A03D8C6B1A2FA5910DE0
+uid                      <YOUR-NAME> (Work) <<YOUR-NAME>@digital.education.gov.uk>
+sub   rsa3072 2021-02-11 [E] [expires: 2023-02-11]
+```
+
+Typing `gpg --list-keys` will repeat the output of `pub`, `uid`, and `sub`
+
+#### Secure password-store with GPG key
+
+```bash
+pass init <GPG-PUB-ID from step above>
+```
+
+Append to `~/.bashrc` or `~/.zshrc`
+```
+export AWS_VAULT_PASS_PREFIX=aws-vault
+export AWS_VAULT_BACKEND=pass
+```
+
+## Configure AWS CLI with AWS Vault profiles
+
+Edit or create the `~/.aws/config` file, and paste this in, replacing `<YOUR-AWS-USERNAME>` in line 11
+
+```
+[default]
+region = eu-west-2
+output = json
+
+; Add user credentials with `aws-vault add tvroot`
+[profile tvroot]
+include_profile = tvcommon
+
+; The "tvcommon" profile here operates as a "config fragment" rather than a profile
+[profile tvcommon]
```

Comment:
I think we shouldn't use `include_profile` as it's non standard and users may confuse it with `source_profile`.
We can duplicate source_profile, region and mfa_serial in each profile. It's more verbose but clearer.

Reply:
Added in with the intentional repetition